pkg/typedef/table.go

```diff
@@ -88,3 +88,37 @@ func (t *Table) Init(s *Schema, c QueryCache) {
 	t.queryCache = c
 	t.queryCache.BindToTable(t)
 }
+
+func (t *Table) ValidColumnsForDelete() []int {
+	if t.Columns.Len() == 0 {
+		return nil
+	}
+	validColsLen := t.Columns.Len()
+	validCols := make([]int, 0, validColsLen)
+	for i := 0; i < validColsLen; i++ {
+		validCols = append(validCols, i)
+	}
+	if len(t.Indexes) != 0 {
```

Comment:
This is very confusing, please make a list of `occupied`/`forbidden` columns first